lapis/job_io/htcondor.py

```diff
@@ -31,36 +32,55 @@ def htcondor_job_reader(
         "DiskUsage_RAW": 1.024 / 1024 / 1024,
     },
 ):
-    htcondor_reader = csv.DictReader(iterable, delimiter=" ", quotechar="'")
+    try:
+        htcondor_reader = json.load(iterable)
+    except ValueError:
+        htcondor_reader = csv.DictReader(iterable, delimiter=" ", quotechar="'")
+    else:
+        logging.getLogger("implementation").error(
+            "Invalid input file %s. Job input file can not be read." % iterable.name
+        )
 
-    for row in htcondor_reader:
-        if float(row[used_resource_name_mapping["walltime"]]) <= 0:
+    for entry in htcondor_reader:
+        if float(entry[used_resource_name_mapping["walltime"]]) <= 0:
             logging.getLogger("implementation").warning(
-                "removed job from htcondor import (%s)", row
+                "removed job from htcondor import (%s)", entry
             )
             continue
         resources = {}
         for key, original_key in resource_name_mapping.items():
             try:
-                resources[key] = float(row[original_key]) * unit_conversion_mapping.get(
-                    original_key, 1
-                )
+                resources[key] = float(
+                    entry[original_key]
+                ) * unit_conversion_mapping.get(original_key, 1)
             except ValueError:
                 pass
+
         used_resources = {
             "cores": (
-                (float(row["RemoteSysCpu"]) + float(row["RemoteUserCpu"]))
-                / float(row[used_resource_name_mapping["walltime"]])
+                (float(entry["RemoteSysCpu"]) + float(entry["RemoteUserCpu"]))
+                / float(entry[used_resource_name_mapping["walltime"]])
             )
             * unit_conversion_mapping.get(used_resource_name_mapping["cores"], 1)
         }
         for key in ["memory", "walltime", "disk"]:
             original_key = used_resource_name_mapping[key]
             used_resources[key] = float(
-                row[original_key]
+                entry[original_key]
             ) * unit_conversion_mapping.get(original_key, 1)
+
+        try:
+            resources["inputfiles"] = {
+                file["filename"]: {
+                    "filesize": file["filesize"],
+                    "usedsize": file["usedsize"],
+                }
+                for file in entry["Inputfiles"]
```

Comment:
I replaced the If-clause by try-except:
`    try:
        htcondor_reader = json.load(iterable)
    except ValueError:
        htcondor_reader = csv.DictReader(iterable, delimiter=" ", quotechar="'")
    else:
        logging.getLogger("implementation").error(
            "Invalid input file %s. Job input file can not be read." % iterable.name
        )`
The except part is accessed correctly, but the resulting htcondor_reader is empty. 
trying on csv files first and handling json files in the except part is not an option as the csv-Reader is able to read something from the json file. The exceptions occur later in that case.

Reply:
Concerning the `inputfiles` dictionary: I'm unsure whether the information (`filename`, `total filesize`) actually represent what the scheduler sees (as far as I know HTCondor knows nothing about filesizes etc.). So in case of two dictionaries the `resources` dictionary would only contain the filenames (probably as a list) while the `used_resources` dictionary would contain nested dictionaries where the filenames are dictionary keys. In this way the data is structured differently in the `resources`/`used_resources` dictionaries and I'm not sure whether this might at some point cause problems. So I thought it might be less confusing if we use only one dictionary (the `used_resources' dictionary) and only use the dictionary keys for everything concerning the scheduler.

Reply:
I think that is actually an interesting point: What if the scheduler *would have* access to file sizes? Would this be useful, e.g. to guess at high/low I/O jobs? Could this be worthwhile for the initial job-placement when caches are empty? It's something that seems worth investigating -- if the scheduler doesn't benefit, we can still ignore it like most of the ClassAd content in reality.

In principle, the HTCondor scheduler doesn't understand CPU, memory, and disk either, it just knows resources, classads and expressions. A submission tool like grid-control would know the file size anyway, and could add it to the file "list". (Newstyle) ClassAds do support ``dict`` like containers (nested ClassAds).

So I would strongly recommend to split ``inputfiles`` across the existing ``resources``/``used_resources`` dicts.

Reply:
If it is possible that the scheduler could have access to file sizes via grid-control I agree with you and I'll change it.